File: src/main/java/com/gym/fitmentor/users/coach/domain/CoachService.java
package com.gym.fitmentor.users.coach.domain;

import com.gym.fitmentor.users.coach.infrastructure.CoachJPA;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

/**
 * Service Interface for managing {@link CoachJPA}.
 */
public interface CoachService {
    /**
     * Save a coach.
     *
     * @param coach the entity to save.
     * @return the persisted entity.
     */
    void save(Coach coach);

    /**
     * Get all the coaches.
     *
     * @return the list of entities.
     */
    List<Coach> findAll();

    /**
     * Get the "id" coach.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    Optional<Coach> findOne(UUID id);

    /**
     * Delete the "id" coach.
     *
     * @param id the id of the entity.
     */
    void delete(UUID id);
}

File: src/main/java/com/inigo/fitmentor/coach/domain/CoachService.kt
package com.inigo.fitmentor.coach.domain

import com.inigo.fitmentor.shared.domain.CoachId

/**
 * Service Interface for managing [com.inigo.fitmentor.client.infrastructure.ClientJpa].
 */
interface CoachService {
    fun save(coach: Coach): Coach

    fun findByCoachId(id: CoachId): Coach?
    fun findAll(): List<Coach>
    fun existsUser(coach: Coach): Boolean
    fun existsCoach(root: Coach): Boolean

}
